base/deprecated.jl

```diff
@@ -1499,4 +1499,12 @@ end
 # Calling promote_op is likely a bad idea, so deprecate its convenience wrapper promote_eltype_op
 @deprecate promote_eltype_op(op, As...) promote_op(op, map(eltype, As)...)
 
+# These conversions should not be defined, see #19896
+Base.convert{T<:Number}(::Type{T},x::Dates.Period) = error("conversion between Periods and Numbers is not defined")
+Base.convert{T<:Dates.Period}(::Type{T},x::Real) = error("conversion between Periods and Numbers is not defined")
+@deprecate convert{R<:Real}(::Type{R},x::Dates.DateTime) R(Dates.datetime2rata(x))
+@deprecate convert{R<:Real}(::Type{R},x::Dates.Date)     R(Dates.datetime2rata(x))
+@deprecate convert(::Type{Dates.DateTime}, x::Real) Dates.rata2datetime(x)
+@deprecate convert(::Type{Dates.Date}, x::Real)     Dates.Date(Dates.rata2datetime(x))
```

Comment:
Shouldn't we also have:

```julia
@deprecate convert(::Type{Dates.DateTime}, x::Real) Dates.DateTime(Dates.UTM(x))
```

The current conversion from Real to DateTime uses Rata Die days instead of Rata Die milliseconds.